masonry/src/contexts.rs

```diff
@@ -403,6 +415,34 @@ impl<'a> MutateCtx<'a> {
     }
 }
 
+// --- MARK: WIDGET_REF ---
+// Methods to get a child WidgetRef from a parent.
+impl<'w> QueryCtx<'w> {
+    /// Return a [`WidgetRef`] to a child widget.
+    pub fn get(self, child: WidgetId) -> WidgetRef<'w, dyn Widget> {
```

Comment:
In a previous comment I wrote:

> `QueryCtx::get` should perhaps take a `&WidgetPod`, like how `MutateCtx::get_mut` takes a `&mut WidgetPod`. It currently takes a `WidgetId`, because `Widget::children_ids` returns just that: ids, not pods. Maybe we can construct `WidgetPod` in the default `Widget::get_child_at_pos` method implementation by assuming `WidgetPodInner::Inserted`.

Is it fine to assume `WidgetPodInner::Inserted`? In that case, we need to add a pub(crate) method that allows constructing a pod that way.

Reply:
Oh, you're right, I forgot about that.

I think taking an id is fine. The other solutions come with their own set of problems.

Reply:
See also https://xi.zulipchat.com/#narrow/stream/317477-masonry/topic/.60Widget.3A.3Achildren_ids.60.20vs.20.60Widget.3A.3Achildren_pods.60.

Reply:
Whoops, at the same time you posted this, I created the Zulip thread. Let's leave it as it is for this PR, and if we decide to, we can do a follow-up changing the signature.